srfi-tools/generate.scm

```diff
@@ -5,6 +5,169 @@
+      To subscribe to the list, follow <a href=\"https://srfi.schemers.org/srfi-list-subscribe.html\">these instructions</a>.
+      You can access previous messages via the mailing list <a href=\"https://srfi-email.schemers.org/srfi-~a/\">archive</a>.
+    </p>
+
+    <ul>
+      <li>Received: ~a</li>
+~a
+      ~a
+    </ul>
+
+<h2 id=\"abstract\">Abstract</h2>
+
+~a
+
+~a
+
+<h2 id=\"copyright\">Copyright</h2>
+<p>&copy; ~a ~a</p>
+
+<p>
+~a
+</p>
+<p>
+~a
+</p>
+<p>
+~a
+</p>
+
+  <hr>
+  <address>Editor: <a href=\"mailto:srfi-editors+at+srfi+dot+schemers+dot+org\">~a</a></address>
+  </body>
+</html>\n")
+
+;; Generate srfi-NNN.html given its data, abstract HTML, and content HTML.
+(define (write-single-srfi-document keyword-options dir abstract content)
+  (let ((fname (string-append "srfi-"
+                              (number->string (srfi-number keyword-options))
+                              ".html")))
+    (call-with-output-file (path-append dir fname)
+      (lambda (port)
+        (display
+         (format srfi-html-template
+                 (parse-year (srfi-date-to-show keyword-options))
+                 (srfi-format-authors (srfi-authors keyword-options))
+                 (srfi-number keyword-options)
+                 (srfi-title keyword-options)
+                 (srfi-number keyword-options)
+                 (srfi-title keyword-options)
+                 (srfi-format-authors (srfi-authors keyword-options))
+                 (srfi-status-string keyword-options)
+                 (srfi-number keyword-options)
+                 (srfi-number keyword-options)
+                 (srfi-number keyword-options)
+                 (srfi-draft-date keyword-options)
+                 ""
+                 (if (eq? (srfi-status keyword-options) 'final)
+                     (string-append "<li>Finalized: "
+                                    (srfi-done-date keyword-options)
+                                    "</li>")
+                     "")
+                 abstract
+                 content
+                 (parse-year (srfi-date-to-show keyword-options))
+                 (srfi-format-authors (srfi-authors keyword-options))
+                 (first (srfi-license-sections))
+                 (second (srfi-license-sections))
+                 (third (srfi-license-sections))
+                 srfi-editor)
+         port)))))
+
+(define (parse-year str)
+  (string-take str 4))
+
+(define (sxml-rest elem proc)
+  (call/cc
+   (lambda (k)
+     (let walk ((elem elem))
+       (cond ((not (pair? elem)) '())
+             ((equal? '@ (car elem)) '())
+             (else (let* ((idx (list-index proc elem))
+                          (matches (if idx (drop elem idx) '())))
+                     (if (not (null? matches))
+                         (k (cdr matches))
+                         (for-each walk (sxml-body elem))))))))))
```

Comment:
This utility deas with SXML generally. It could go into the `(srfi-tools private sxml)` library.